File: ChatSupport/src/main/java/com/vantagecircle/chatapp/core/FileModel.java
package com.vantagecircle.chatapp.core;

import com.google.firebase.storage.FileDownloadTask;
import com.google.firebase.storage.StorageMetadata;
import com.google.firebase.storage.StreamDownloadTask;
import com.google.firebase.storage.UploadTask;

/**
 * Created by bapidas on 04/08/17.
 */

public class FileModel {
    private FileDownloadTask.TaskSnapshot fileDownloadSnap;
    private UploadTask.TaskSnapshot uploadTaskSnap;
    private StorageMetadata storageMetadata;
    private Exception exception;
    private String extraString;

    public FileDownloadTask.TaskSnapshot getFileDownloadSnap() {
        return fileDownloadSnap;
    }

    public void setFileDownloadSnap(FileDownloadTask.TaskSnapshot fileDownloadSnap) {
        this.fileDownloadSnap = fileDownloadSnap;
    }

    public UploadTask.TaskSnapshot getUploadTaskSnap() {
        return uploadTaskSnap;
    }

    public void setUploadTaskSnap(UploadTask.TaskSnapshot uploadTaskSnap) {
        this.uploadTaskSnap = uploadTaskSnap;
    }

    public StorageMetadata getStorageMetadata() {
        return storageMetadata;
    }

    public void setStorageMetadata(StorageMetadata storageMetadata) {
        this.storageMetadata = storageMetadata;
    }

    public Exception getException() {
        return exception;
    }

    public void setException(Exception exception) {
        this.exception = exception;
    }

    public String getExtraString() {
        return extraString;
    }

    public void setExtraString(String extraString) {
        this.extraString = extraString;
    }
}

File: app/src/main/java/com/bapidas/chattingapp/data/core/model/FileModel.kt
package com.bapidas.chattingapp.data.core.model

import com.google.firebase.storage.FileDownloadTask
import com.google.firebase.storage.StorageMetadata
import com.google.firebase.storage.UploadTask

/**
 * Created by bapidas on 04/08/17.
 */
data class FileModel(
        var fileDownloadSnap: FileDownloadTask.TaskSnapshot? = null,
        var uploadTaskSnap: UploadTask.TaskSnapshot? = null,
        var storageMetadata: StorageMetadata? = null,
        var exception: Exception? = null,
        var extraString: String = ""
)